src/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,208 @@
+
+
+#include <emscripten/emscripten.h>
+#include <emscripten/val.h>
+#include <emscripten/bind.h>
+
+#include <string>
+#include <iostream>
+
+#include "ffi/ir_stream/decoding_methods.hpp"
+#include "spdlog_with_specializations.hpp"
+#include "TimestampPattern.hpp"
+#include "streaming_compression/zstd/Decompressor.hpp"
+#include "ir/LogEventDeserializer.hpp"
+#include "ffi/encoding_methods.hpp"
+
+#ifdef __cplusplus
+#define EXTERN extern "C"
+#else
+#define EXTERN
+#endif
```

Comment:
This macro is not used in this file, right?
And it might be worth to have a dedicated file for it if we need it in the future. We have sth similar in Go ffi: https://github.com/y-scope/clp-ffi-go/blob/main/cpp/src/ffi_go/api_decoration.h

Reply:
Right, this was some experiment code to get the compilation working and now it can be removed.

Reply:
True. Since this is the first PR, I think it would make sense to have a clear code structure :D